AirSkyBoat/AirSkyBoat
# Fix Lockstyle of Suit style equipment

**_I affirm:_**


## Please enter a player-facing description

When lockstyled - Suit style equipment like the Mandragora Suit will now properly override all legs/feet/hands appearances.  (Tiberon)
A fix is coming soon™ for when suits are equipped and not lockstyled.

## What does this pull request do? (Please be technical)

1. Created a new db column `rslotlook` in `item_equipments.sql`.  This repesents a bitmask of the `SLOT TYPE` which are overrided by a piece of gear **for appearance**.  This is distinctly different from the `rslot` column which is the bit mask for equipment slots which are exclusive with this item being equipped.
2. Populated this new column by: 
     freshly importing the `item_equipments.sql`
     adding the column via heidiSQL
     running an update to populate rslotlook from rslot as a starting point
     testing each NQ version of an item which had an rslot value to find the gaps
     updating each gap item - which was generally all the "suits" aside from the moogle suit
3. Because I populated a new column - the PR refuses to show every single line of the sql - I've included a table below which summarizes which items have a delta in the new column. 
4. I searched all of the ASB sql update modules - there were no inserts (good, there shouldnt be) so no additional changes needed for the column addition.
5. Plumbed this new column value into CItem, exposing getter and setter functions
6. Updated the recently added lockstyle helper function `UpdateRemovedSlots` to use the newly added getter

## Steps to test these changes
1. equip something bulky/noticeable in head/body/hands/legs/feet.  I used the mnk relic set, melee.  Its bulky ish and purple
2. Add an item from the below list and put into a lock style.
```
25715	korrigan_suit
25726	kupo_suit
25756	wyrmking_suit
25757	wyrmking_suit_+1
26954	behemoth_suit
26955	behemoth_suit_+1
26963	onca_suit
27854	mandra._suit
27855	mandra._suit_+1
27866	goblin_suit
27867	g._moogle_suit
27906	chocobo_suit_+1
27911	chocobo_suit
```
3. Add equipment into the lockstyle in any other slot that will be overriden like hands/legs/feet
example: 
![image](https://github.com/AirSkyBoat/AirSkyBoat/assets/105882290/fbaf19f5-411f-4530-b238-b690ff860903)
4. apply the lock style - and note that there is no "underneath equipment" clipping through
5. remove lockstyle, ensure nothing is missing (like the characters hands, legs, or feet

Items impacted by this change:
![image](https://github.com/AirSkyBoat/AirSkyBoat/assets/105882290/4406643d-2070-4246-b9fb-d5504d615cb3)


Failure examples
![image](https://github.com/AirSkyBoat/AirSkyBoat/assets/105882290/72953091-4a6d-4f11-8f1e-98fc5ac135ce)

Success examples:
![image](https://github.com/AirSkyBoat/AirSkyBoat/assets/105882290/4944b595-0f0a-4910-970c-ec1f7abd20e6)

## Follow Ups
- Back port this to LSB
- Fix the look on equip - a bit tricky because we set the look in around 8 places in code.

## Special Deployment Considerations

need to run item_equipments.sql - it will add a new column.
